media-video/motion/motion-4.1.1-r2.ebuild

```diff
@@ -0,0 +1,99 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+inherit autotools readme.gentoo-r1 systemd user
+
+DESCRIPTION="A software motion detector"
+HOMEPAGE="https://motion-project.github.io"
+SRC_URI="https://github.com/Motion-Project/${PN}/archive/release-${PV}.tar.gz -> ${PN}-release-${PV}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~hppa ~ia64 ~ppc64 ~sparc ~x86"
+IUSE="ffmpeg libav mariadb mmal mysql postgres sqlite v4l webp"
+REQUIRED_USE="
+	?? ( ffmpeg libav )
+	?? ( mariadb mysql )
+"
+
+RDEPEND="
+	virtual/jpeg:=
+	ffmpeg? ( media-video/ffmpeg:0= )
+	libav? ( media-video/libav:= )
+	mmal? ( media-libs/raspberrypi-userland )
+	mysql? ( dev-db/mysql-connector-c )
+	mariadb? ( dev-db/mariadb-connector-c )
+	postgres? ( dev-db/postgresql:= )
+	sqlite? ( dev-db/sqlite:3 )
+	webp? ( media-libs/libwebp:= )
+"
+DEPEND="${RDEPEND}
+	v4l? (
+		media-libs/libv4l
+		virtual/os-headers
+	)
+"
+
+PATCHES=(
+	"${FILESDIR}"/${PV}/Fix-build-errors-with-FFmpeg-4.0.patch
+)
+
+DISABLE_AUTOFORMATTING="yes"
+DOC_CONTENTS="
+You need to setup /etc/motion/motion.conf before running motion for the
+first time. You can use /etc/motion/motion-dist.conf as a template.
+Please note that the 'daemon' and 'process_id_file' settings are
+overridden by the bundled OpenRC init script and systemd unit where
+appropriate.
+
+To install motion as a service, use:
+rc-update add motion default # with OpenRC
+systemctl enable motion.service # with systemd
+"
+
+S="${WORKDIR}"/${PN}-release-${PV}
+
+pkg_setup() {
+	enewuser ${PN} -1 -1 -1 video
+}
+
+src_prepare() {
+	default
+	eautoreconf
+}
+
+src_configure() {
+	econf \
+		$(use_with ffmpeg) \
+		$(use_with libav ffmpeg) \
+		$(use_with mariadb mysql) \
```

Comment:
1. Are you really convinced `--with-mysql` will correctly use libmariadb* and not simultaneously installed mysql-connector-c?
2. This makes USE=mariadb do nothing, as you pass second `--with-mysql` or `--without-mysql` below, so it will override it.

Reply:
The motion configure script only provides a `--with-mysql` option that first checks for mariadb and then falls-back to a mysqlclient check if mariadb is not found. Since there is no way provided by the mortion configure to explicitly select between mariadb or mysqlclient, should I remove the mysql USE flag and leave it solely with a mariadb dependency on the mariadb USE flag?

Reply:
I'm sorry but I've just read that configure.ac, and I need to find some happy place for myself ;-P. For a start, they're calling pkg-config manually which is horribly broken and skips all the good stuff autoconf has pkg-config, and which would certainly let us solve this. In this case, I would indeed suggest just forcing mariadb as the only option given upstream's complete lack of good taste and build system skills.

Long term, it would be nice to rewrite the configure script to be sane and use standard macros. If you'd like to undergo that work, feel free to ask me for early review before submitting upstream.

Reply:
Yes, I agree their configure script is in need of some updates and improvements. I believe it's scheduled for an overhaul for the next release after 4.2 is out soon, so I'll keep and eye on their plans and perhaps help out with improvements to clean up the existing code.

For now, I'll retire the mysql flag for the Gentoo motion package and replace it with the mariadb flag. That should make it obvious to users that mariadb is used for MySQL support in this Gentoo package.

Reply:
I'm sorry to say but this package is even more broken than we suspected. It indeeds seems to prefer mariadb but uses `AC_CHECK_LIB` on libmysqlclient. So:
* if mariadb + mysql connectors are both installed, it links to mariadb,
* if mariadb is only installed, it doesn't use mariadb.

It might work with `dev-db/mariadb-connector-c[mysql-compat]` but I haven't tested that.

Reply:
You are right, since the `AC_CHECK_LIB` on mysqlclient in the motion configure line prevents it from using mariadb, we need `dev-db/mariadb-connector-c[mysql-compat]` as a dependency. I tested it with the `mysql-compat` flag enabled just now and it does appear to detect MySQL support in the motion configure script now.

I'll update this version of the motion ebuild to add the `dev-db/mariadb-connector-c[mysql-compat]` dependency, but it's yet another indication that the motion configure script needs to be overhauled soon.